Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -187,31 +180,21 @@ public final class SourceKitServer {
 
   /// Caches which workspace a document with the given URI should be opened in.
   /// Must only be accessed from `queue`.
-  private var uriToWorkspaceCache: [DocumentURI: WeakWorkspace] = [:] {
-    didSet {
-      dispatchPrecondition(condition: .onQueue(queue))
-    }
-  }
+  private var uriToWorkspaceCache: [DocumentURI: WeakWorkspace] = [:]
 
-  /// Must only be accessed from `queue`.
   private var workspaces: [Workspace] = [] {
     didSet {
-      dispatchPrecondition(condition: .onQueue(queue))
       uriToWorkspaceCache = [:]
     }
   }
 
   /// **Public for testing**
   public var _workspaces: [Workspace] {
```

Comment:
Do we still need both `workspaces` and `_workspaces` now? Are there any accesses other than for setting it in tests? If not you could just remove `_workspaces`, and mark the new setter function as SPI.

Reply:
Sounds good 👍